src/engine/threaded_engine.h

```diff
@@ -476,6 +507,9 @@ class ThreadedEngine : public Engine {
    */
   std::mutex finished_m_;
   std::condition_variable finished_cv_;
+  /*! \brief exception_ptr associated with the engine,
+   * which is used to throw exception in waitall */
+  std::exception_ptr global_ex_ptr;
```

Comment:
ex_ptr is a bad name. class members should end with `_`.

Reply:
Changed!